src/firstboot/firstboot.c

```diff
@@ -145,14 +145,19 @@ static int prompt_loop(int rfd, const char *text, char **l, unsigned percentage,
                 _cleanup_free_ char *p = NULL;
                 unsigned u;
 
-                r = ask_string(&p, strv_isempty(l) ? "%s %s (empty to skip): "
-                                                   : "%s %s (empty to skip, \"list\" to list options): ",
-                               special_glyph(SPECIAL_GLYPH_TRIANGULAR_BULLET), text);
+                r = ask_string(&p, strv_isempty(l) ? "%s %s (empty to default to \"%s\", \"-\" to skip): "
+                                                   : "%s %s (empty to default to \"%s\", \"-\" to skip, \"list\" to list options): ",
```

Comment:
I'd like to keep the behavior for an empty string. How about the following?
- empty string -> skip
- "default" -> explicitly set default string
- "list" -> list options

Reply:
I was mostly following what was suggested in this comment https://github.com/systemd/systemd/pull/31154#issuecomment-1946056086 (Point 1 and 6).

I am not a fan of using the literal "default" to check for that, but I also don't know what really could be used for that other than how it currently is.

Reply:
Not a strong opinion, but I'd like to keep the behavior for an empty string, as explicitly setting default value is typically not necessary.

"-" may be used for the literal for applying default explicitly, if "default" is not good.

Reply:
> IIRC, @poettering also commented similar in the above (not searched).

He did, just he didn't follow up on my clarification.

I'd say that that commit is still something that needs more work and discussion, as mentioned above the defaults/fallbacks used currently could be better than just the compile time default.

Reply:
@poettering Could you comment on this?

There are two reasons I want to swap an empty string and "-":
- (already commented in the above, but again) explicitly setting default value is typically not necessary
- We usually handle an empty string as doing compiled time default, especially in unit files or .conf files. So, this conflicts with usual behaviors of other components.

> > IIRC, @poettering also commented similar in the above (not searched).
> 
> He did, just he didn't follow up on my clarification.

Maybe https://github.com/systemd/systemd/pull/33226#discussion_r1648902754 ? But I cannot follow your reply.

> I'd say that that commit is still something that needs more work and discussion, as mentioned above the defaults/fallbacks used currently could be better than just the compile time default.

Maybe. But, even so, I'd like to keep empty to do nothing.

Reply:
> Maybe https://github.com/systemd/systemd/pull/33226#discussion_r1648902754 ? But I cannot follow your reply.

I was talking about this comment https://github.com/systemd/systemd/pull/33226#discussion_r1649455325 (forgot I marked that discussion as resolved)

My arguments were:

1. These prompts need to be opted in to with either `--prompt` or the respective `--prompt-*` option (although with the `sd-firstboot.service` change this argument is debatable)
1. These few configuration files writen in this service aren't really something that (a) shouldn't exist on a system (this already caused us to copy the default file from `/usr/share/factory/` https://github.com/systemd/systemd/pull/31154#issuecomment-1946174780) and (b) is highly system specific and not distribution specific and changes to the values from a distribution side could be surprising.

Reply:
> - empty string -> skip
> - "default" -> explicitly set default string
> - "list" -> list options

@yuwata I just had the thought: what about instead of "default" using `<tab>`? I feel that makes a lot of sense since it's already use for completion in shells, so it would also follow that it would work here.

Reply:
> > * empty string -> skip
> > * "default" -> explicitly set default string
> > * "list" -> list options
> 
> @yuwata I just had the thought: what about instead of "default" using `<tab>`? I feel that makes a lot of sense since it's already use for completion in shells, so it would also follow that it would work here.

Using tab sounds OK to me.

Reply:
I was just looking into this. Either I was missing some existing APIs or it would need to be implemented (as just checking for a returned tab is rather ugly on the user side), at which point it might just make more sense to skip this entire "default selection" stuff and implement full tab completion.